cmd/kubermatic-installer/quickstart_kind_resources.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2023 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilintstr "k8s.io/apimachinery/pkg/util/intstr"
+
+	providerconfig "github.com/kubermatic/machine-controller/pkg/providerconfig/types"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+)
+
+var kindConfigContent = `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+ - role: control-plane
+ - role: worker
+   extraPortMappings:
+   # nodeport-proxy for user-cluster control-planes
+   - containerPort: 30652
+     hostPort: 6443
+   - containerPort: 32121
+     hostPort: 8088
+   # ingress controller for kubermatic api and dashboard 
+   - containerPort: 31514
+     hostPort: 80
+   - containerPort: 32394
+     hostPort: 443`
+
+func ptr[T any](v T) *T {
+	return &v
+}
+
+var kindKubermaticNamespace = corev1.Namespace{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "kubermatic",
+	},
+}
+
+var kindIngressControllerNamespace = corev1.Namespace{
+	ObjectMeta: metav1.ObjectMeta{
+		Name: "nginx-ingress-controller",
+	},
+}
+
+var kindStorageClass = storagev1.StorageClass{
```

Comment:
might be worth exploring how to add this into master install stack instead of harcoding here?
https://github.com/kubermatic/kubermatic/blob/2d6e26802c84de9d335ade02f836194da6b16967/pkg/install/stack/kubermatic-master/stack.go#L151

Reply:
I would like to leave this as todo for follow-up task